### PR TITLE
Parse properties from `maven.config` file

### PR DESCRIPTION
**Pull request by Laurens-W**
## What's changed?

Also parse `-D` style properties from `maven.config`.

## Anything in particular you'd like reviewers to focus on?

Test coverage

## Anyone you would like to review specifically?

@timtebeek @greg-at-moderne 

## Any additional context

For now we treat the `maven.config` as a read only file, so it is correct that we cannot write an updated version back to it.

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've read and applied the [recipe conventions and best practices](https://docs.openrewrite.org/authoring-recipes/recipe-conventions-and-best-practices)
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

**Comment by timtebeek** [comment]
Thanks! I think it's more so a problem when it comes to the Maven plugin and building LSTs; did you try that for a sample project?

**Comment by Laurens-W** [comment]
> Thanks! I think it's more so a problem when it comes to the Maven plugin and building LSTs; did you try that for a sample project?

Building a project locally with the minimal example works as intended, as well as an OS project I found

File: rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java [hunk 1]
```diff
@@ -3707,4 +3708,98 @@ void systemPropertyTakesPrecedence() {
...
+              
+                  <groupId>com.example</groupId>
+                  <artifactId>my-app</artifactId>
+                  <version>${revision}</version>
+              </project>
+              """,
+            spec -> spec.afterRecipe(p -> {
+                var results = p.getMarkers().findFirst(MavenResolutionResult.class).orElseThrow();
+                assertThat(results.getPom().getVersion()).isEqualTo("${revision}");
+                assertThat(results.getPom().getProperties().get("revision")).isEqualTo("1.0.0");
```
Comment (jevanlingen) [inline rewrite-maven/src/test/java/org/openrewrite/maven/MavenParserTest.java:3734]: I am starting to think the assumption of having the `maven.config` as extra properties is not right. The information of the `text` block is another LST, we don't merge those together. Same applies for Gradle:

```java
@Test
void property() {
    rewriteRun(
      properties("junitVersion=4.13", spec -> spec.path("gradle.properties")),
      buildGradle(
        """
          plugins { id 'java-library' }
          repositories { mavenCentral() }
          dependencies { testImplementation "junit:junit:${junitVersion}" }
          """,
        spec -> spec.afterRecipe(cu -> cu.getMarkers().getMarkers().get(0))
      )
    );
}
```

That marker would show:

```
org.codehaus.groovy.syntax.SyntaxException: [Static type checking] - The variable [junitVersion] is undeclared.
 @ line 10, column 39.
  org.codehaus.groovy.control.ErrorCollector.addErrorAndContinue(ErrorCollector.java:84)
````

This is also in line with current recipe's. For example the _DependencyVulnerabilityCheck_ recipe, which I have now extended knowledge about, also just treats the properties LST as a different input.

File: rewrite-maven/src/test/java/org/openrewrite/maven/UpgradeDependencyVersionTest.java [hunk 2]
```diff
@@ -2008,4 +2009,45 @@ void exactVersionMissingInMavenMetadata() {
           )
         );
     }
+
+    @Test
+    void upgradeVersionWithGroupIdAndArtifactIdDefinedAsMavenConfigProperty() {
+        rewriteRun(
```
Comment (jevanlingen) [inline rewrite-maven/src/test/java/org/openrewrite/maven/UpgradeDependencyVersionTest.java:2015]: This test fails at the moment, because the properties in the text LST are not recognised:

image

**Review by MBoegers — CHANGES_REQUESTED** [review]
Could you please add a test for an empty config file?

**Comment by Laurens-W** [comment]
> Could you please add a test for an empty config file?

Like the one I added?
